src/state/migrations.test.ts

```diff
@@ -0,0 +1,154 @@
+    callResults: {},
+  },
+  routingApi: {
+    config: {
+      focused: true,
+      keepUnusedDataFor: 60,
+      middlewareRegistered: true,
+      online: true,
+      reducerPath: 'routingApi',
+      refetchOnFocus: false,
+      refetchOnMountOrArgChange: false,
+      refetchOnReconnect: false,
+    },
+    mutations: {},
+    provided: {},
+    queries: {},
+    subscriptions: {},
+  },
+  routingApiV2: {
+    config: {
+      focused: true,
+      keepUnusedDataFor: 60,
+      middlewareRegistered: true,
+      online: true,
+      reducerPath: 'routingApiV2',
+      refetchOnFocus: false,
+      refetchOnMountOrArgChange: false,
+      refetchOnReconnect: false,
+    },
+    mutations: {},
+    provided: {},
+    queries: {},
+    subscriptions: {},
+  },
+  wallets: {
+    connectedWallets: [],
+    switchingChain: false,
+  },
+}
+
+describe('redux migrations', () => {
+  let store: Store
+
+  beforeEach(() => {
+    localStorage.clear()
+    // We need to re-create the store before each test so it starts with undefined state.
+    store = createDefaultStore()
+  })
+
+  it('clears legacy redux_localstorage_simple values during the initial migration', async () => {
+    localStorage.setItem('redux_localstorage_simple_transactions', JSON.stringify({ test: 'transactions' }))
+    localStorage.setItem('redux_localstorage_simple_user', JSON.stringify({ test: 'user' }))
+    localStorage.setItem('redux_localstorage_simple_lists', JSON.stringify({ test: 'lists' }))
+
+    persistStore(store)
+    // wait for the migration to complete
+    await new Promise((resolve) => setTimeout(resolve, 0))
+
+    expect(localStorage.getItem('redux_localstorage_simple_transactions')).toBeNull()
+    expect(localStorage.getItem('redux_localstorage_simple_user')).toBeNull()
+    expect(localStorage.getItem('redux_localstorage_simple_lists')).toBeNull()
+
+    const state = store.getState()
+    expect(state).toMatchObject({
+      ...defaultState,
+      // These are migrated values.
+      lists: {
+        test: 'lists',
+      },
+      transactions: {
+        test: 'transactions',
+      },
+      user: {
+        test: 'user',
+      },
+    })
+  })
+
+  it('initial state with no previous persisted state', async () => {
+    persistStore(store)
+    // wait for the migration to complete
+    await new Promise((resolve) => setTimeout(resolve, 0))
+    const state = store.getState()
+    expect(state).toMatchObject(defaultState)
+  })
+
+  it('migrates from a previous version of the state type', async () => {
+    localStorage.setItem(
+      'persist:root',
+      JSON.stringify({
+        user: { ...initialUserState, test: 'user' },
+        transactions: initialTransactionsState,
+        lists: initialListsState,
+        _persist: { version: -1 },
+      })
+    )
+
+    persistStore(store)
+    // wait for the migration to complete
+    await new Promise((resolve) => setTimeout(resolve, 0))
+
+    const state = store.getState()
```

Comment:
yeah, i spent alot of time trying to verify the persisted data but i couldn't get localStorage working in jest . i need to mock it somehow because redux-persist doesn't seem to write to jest's localStorage by default.

in the meantime, i've verified the persisted data with manual testing (see screenshots in PR description)

Reply:
jest runs in node, so you'd need a mock regardless. you can use a polyfill if you can find one on npm, if that would increase your confidence.